packages/ses/src/tame-faux-data-properties.js

```diff
@@ -0,0 +1,131 @@
+import {
+  getOwnPropertyDescriptor,
+  apply,
+  defineProperty,
+  toStringTagSymbol,
+} from './commons.js';
+
+const throws = thunk => {
+  try {
+    thunk();
+  } catch (er) {
+    return true;
+  }
+  return false;
+};
+
+/**
+ * Exported for convenience of unit testing. Harmless, but not expected
+ * to be useful by itself.
+ *
+ * @param {any} obj
+ * @param {string|symbol} prop
+ * @param {any} expectedValue
+ * @returns {boolean}
+ */
+export const tameFauxDataProperty = (obj, prop, expectedValue) => {
+  if (obj === undefined) {
+    return false;
+  }
+  const desc = getOwnPropertyDescriptor(obj, prop);
+  if (!desc || 'value' in desc) {
+    return false;
+  }
+  const { get, set } = desc;
+  if (typeof get !== 'function') {
+    return false;
+  }
+  if (typeof set !== 'function') {
+    return false;
+  }
+  const observedValue = get();
+  if (observedValue !== expectedValue) {
+    return false;
+  }
+  if (apply(get, obj, []) !== expectedValue) {
+    return false;
+  }
+  const testValue = 'Seems to be a setter';
+  const subject1 = { __proto__: null };
+  apply(set, subject1, [testValue]);
+  if (subject1[prop] !== testValue) {
+    return false;
+  }
+  const subject2 = { __proto__: obj };
+  apply(set, subject2, [testValue]);
+  if (subject2[prop] !== testValue) {
+    return false;
+  }
```

Comment:
Do we care about the setter behavior?

Reply:
Since we're sniffing, to see if the observed accessor seems to be an attempt at similarly enabling property overrides, yes. If the setter does something different, then the accessor is unlikely to be emulating a data property, so we should not assume we can preserve intended semantics if we turn it into an actual data property. Better to fail in permits-intrinsics.js until a developer can look and understand the anomaly.

Resolving.

Reply:
I suspect that this sniff is neither helpful nor harmful: not consequential. Given: we are using this repair only on properties we know are either:

1. not implemented
2. implemented as a data property per prior specification (these were all unshipped)
3. implemented as a getter or setter per to specification
4. implemented to a future specification likely to resemble 2 or 3
5. not implemented to specification

The sniff pertains to the last case. In many places, we assume that if a function exists, it behaves according to specification, like `Map.prototype.set`. All such functions are in the TCB. There are many ways that these functions could misbehave, including:

1. ignoring the subject and entangling the getter and setter over otherwise shared state
2. ignoring the set
3. inconsistent get
4. observable side-effects
5. un-observable side-effects

Reply:
I mostly accept this reasoning. The point is not to catch malice or purposeful misbehavior. The point is to catch cases where the underlying platform changed since that version of SES was written, and we want to catch if it seems to have only been changed in ways we expect. If we're confident enough of that, then we let ses succeed at initializing. We purposely tolerate all additions that we can remove, because we're confident enough that this is symptomatic of purely additive changes, that don't break our safety once removed.

Right now, having the spec define accessor functions is quite novel for tc39. I do not yet feel confident that tc39 will follow the (IMO good) precedent set by the current state of the iterators-helpers proposal. So I want us to continue to fail early when there is a platform change we need to think about before concluding that it does not harm SES's safety.

> nor harmful: not consequential

I interpret this as I'm clear to proceed ;)

Reply:
In light of how this thread has gone, I restored the stricter setter behavior testing. Resolving.